libs/components/molecules/src/lib/interactive-svg/interactive-svg.component.ts

```diff
@@ -167,8 +173,13 @@ export class InteractiveSvgComponent implements OnDestroy {
    * @returns Parent id
    */
   private getId(event: Event): string {
-    const parentEl = (event.target as Element).parentElement as Element;
-    return parentEl.id;
+    const parent = (event.target as Element).parentElement;
+    const grandparent = parent?.parentElement;
+    let id = this.decodeId((event.target as Element).id);
+    if (!id) {
+      id = parent?.id ? parent.id : grandparent?.id || '';
+    }
+    return id;
```

Comment:
Decoding the id should come last. If it is taken from the parent/grandparent it should still be decoded